internal/resources/ibm/resource_instance.go

```diff
@@ -284,7 +296,116 @@ func GetAppIDCostComponents(r *ResourceInstance) []*schema.CostComponent {
 		}
 	} else {
 		costComponent := schema.CostComponent{
-			Name: fmt.Sprintf("Plan: %s", r.Plan),
+			Name:            fmt.Sprintf("Plan: %s", r.Plan),
+			UnitMultiplier:  decimal.NewFromInt(1),
+			MonthlyQuantity: decimalPtr(decimal.NewFromInt(1)),
+		}
+		costComponent.SetCustomPrice(decimalPtr(decimal.NewFromInt(0)))
+		return []*schema.CostComponent{
+			&costComponent,
+		}
+	}
+}
+
+func AppConnectFlowsRunsCostComponent(r *ResourceInstance) *schema.CostComponent {
+	var q *decimal.Decimal
+	if r.AppConnect_ThousandRuns != nil {
+		q = decimalPtr(decimal.NewFromFloat(*r.AppConnect_ThousandRuns))
+	}
+	return &schema.CostComponent{
+		Name:            "Flow runs",
+		Unit:            "1k runs",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: q,
+		ProductFilter: &schema.ProductFilter{
+			VendorName: strPtr("ibm"),
+			Region:     strPtr(r.Location),
+			Service:    &r.Service,
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "planName", Value: &r.Plan},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			Unit: strPtr("ITEMS_PER_MONTH"),
+		},
+	}
+}
+
+func AppConnectEgressCostComponent(r *ResourceInstance) *schema.CostComponent {
+	var q *decimal.Decimal
+	if r.AppConnect_GigabyteTransmittedOutbounds != nil {
+		q = decimalPtr(decimal.NewFromFloat(*r.AppConnect_GigabyteTransmittedOutbounds))
+	}
+	return &schema.CostComponent{
+		Name:            "Gigabytes transmitted outbounds",
+		Unit:            "GB",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: q,
+		ProductFilter: &schema.ProductFilter{
+			VendorName: strPtr("ibm"),
+			Region:     strPtr(r.Location),
+			Service:    &r.Service,
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "planName", Value: &r.Plan},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			Unit: strPtr("GIGABYTE_TRANSMITTED_OUTBOUND"),
+		},
+	}
+}
+
+func AppConnectCpuCostComponent(r *ResourceInstance) *schema.CostComponent {
+	var q *decimal.Decimal
+	if r.AppConnect_VCPUHours != nil {
+		q = decimalPtr(decimal.NewFromFloat(*r.AppConnect_VCPUHours))
+	}
+	return &schema.CostComponent{
+		Name:            "VCPU",
+		Unit:            "VCPU hours",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: q,
+		ProductFilter: &schema.ProductFilter{
+			VendorName: strPtr("ibm"),
+			Region:     strPtr(r.Location),
+			Service:    &r.Service,
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "planName", Value: &r.Plan},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			Unit: strPtr("GIGABYTE_TRANSMITTED_OUTBOUND"),
```

Comment:
I think this should be VIRTUAL_PROCESSOR_CORE_HOURS instead

Reply:
yep